### ISSUE TITLE
Error only reported through status handler?

### ISSUE THREAD

**Issue by paulmelis** [ospray/ospray#236]
I was trying to debug why, while setting up a volume, the call to `commit()` was causing my program to exit with code 1. Under gdb it looked like an exception was thrown, but setting an error handler didn't report anything. Only after I had set a *status* handler with `ospDeviceSetStatusFunc()` was the reason revealed:

```
MESSAGE:   ospray::SharedStructuredVolume<float>  ERROR: no transfer function specified.
```

I had tried fiddling with OSPRAY_LOG_LEVEL, OSPRAY_LOG_OUTPUT and OSPRAY_ERROR_OUTPUT, but apprently that also requires that you have set a status handler to get the error.

**Comment by jeffamstutz** [comment]
Hi,

That doesn't sound right and is definitely a bug! Fortunately the bug appears in a place that is a commonly used error reporting function that can instead throw an error (which will get printed with the error message handler).

Sorry about the confusion!

Jeff

**Comment by ingowald** [comment]
Probably the same tbb::exception != runtime_error I ran in last week.

**Comment by jeffamstutz** [comment]
No, ```ManagedObject::exitOnCondition()``` would emit a _status_ message, then call ```std::exit(1)```. Neither of those are preferred: we post error messages by catching thrown exceptions behind the API, and calling ```std::exit(1)``` inside OSPRay doesn't give the application the choice to handle errors a different way.

All of our volume code used the old ```exitOnCondition()``` method of hard exiting when something goes wrong. I've cleaned this up internally (i.e. converted to throwing exceptions that will get reported via the error handler) and will push to ```release-1.5.x``` once our CI passes....it sounds like the start of some fixes for a ```v1.5.1``` patch release.

**Comment by jeffamstutz** [comment]
Status update: we are going to go straight to a ```v1.6.0``` release because we changed the way we build our documentation. FYI this is fixed on ```devel```, which will end up in ```v1.6.0```.
